File: programacao_orientada/cadastro_funcionario_python/classes/Empresa.py
class Empresa:
    def __init__(self):
        self.funcionarios = [] 

    def adicionar_funcionario(self, f):
        self.funcionarios.append(f)

    def listar_funcionarios(self):
        if len(self.funcionarios) == 0:
            print("Nenhum funcionário cadastrado na empresa.\n")
        else:
            for f in self.funcionarios:
                print(f"{f.get_nome()} - {type(f).__name__}")

    def comecar_trabalho(self):
        if len(self.funcionarios) == 0:
            print("Nenhum funcionário para trabalhar hoje!\n")
        else:
            for f in self.funcionarios:
                f.executar_tarefa()

File: programacao_orientada/cadastro_funcionario_java/src/empresa.java
import java.util.ArrayList;

public class empresa {

    private ArrayList<funcionario> funcionarios = new ArrayList<>();

    public void adicionarFuncionario(funcionario f){
        funcionarios.add(f);
    }

    public void listarFuncionarios(){
        if (funcionarios.isEmpty()) {
            System.out.println("Nenhum funcionário cadastrado na empresa.\n");
            return;
        } else {
            for (funcionario f : funcionarios) {
                System.out.println(f.getNome() + " - " + f.getClass());
            }
        }
    }

    public void comecarTrabalho(){
        if (funcionarios.isEmpty()) {
            System.out.println("Nenhum funcionário para trabalhar hoje!\n");
            return;
        } else {
            for (funcionario f : funcionarios) {
                f.executarTarefa();
            }
        }
    }

}
